devito/ir/support/basic.py

```diff
@@ -60,7 +60,7 @@ class IterationInstance(LabeledVector):
     """
 
     def __new__(cls, indexed):
-        findices = tuple(indexed.function.indices)
+        findices = tuple(indexed.function.dimensions)
```

Comment:
can you check ir/clusters/algorithms.py ? there might be another indices->dimensions change required

EDIT: [this](https://github.com/opesci/devito/blob/master/devito/ir/clusters/algorithms.py#L425) and [this](https://github.com/opesci/devito/blob/master/devito/ir/clusters/algorithms.py#L427) (both `f.indices`)

Reply:
everything in ir/cluster/algorithm is after indexing and all Indexed object are working with indices.